docs/docs/ports.md

```diff
@@ -0,0 +1,234 @@
+
+*ports*: The _ports_ array is used to define ports that should be considered as part of a resource offer. It is necessary only to define this array if you are using `HOST` networking and no port mappings are specified.
+
+*protocol*: _Protocol_ specifies the internet protocol to use for a port (e.g. `tcp` or `udp`). This is only necessary as part of a _port mapping_ when using `BRIDGE` mode networking with a Docker container.
+
+*requirePorts*: _requirePorts_ is a property that specifies whether Marathon should specifically look for specified ports in the resource offers it receives. This ensures that these ports are free and available to be bound to on the Mesos agent. This does not apply to `BRIDGE` mode networking.
+
+*servicePort*: A _service port_ is a port used to describe the port that a service should made available at. Marathon does not bind to the service ports specified but will ensure that you cannot have multiple applications that use the same service port running on the same host. Service ports are typically only used by external applications (e.g. HAProxy) to make the application available at the specified port. See [Service Discovery & Load Balancing](service-discovery-load-balancing.html) for more information.
+
+## Random Port Assignment
+
+Using the value 0 for any port settings indicates to Marathon that you would like a random port assignment. However, if containerPort is set to 0 within a portMapping, it is set to the same value as hostPort.
+
+## Environment Variables
+
+Each port value is exposed to the running application instance via environment variables `$PORT0`, `$PORT1`, etc. Each Marathon application is given a single port by default, so `$PORT0` is always available. These variables are available inside a Docker container being run by Marathon too.
+
+## Example Configuration
+
+### Host Mode
+
+Host mode networking is the default networking mode for Docker containers and the only networking mode for non-Docker applications. Note that it not necessary to `EXPOSE` ports in your Dockerfile.
+
+#### Enabling Host Mode
+
+Host mode is enabled by default for containers. If you wish to be explicit, you can also specify it manually through the `network` property:
+
+```json
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "my-image:1.0",
+      "network": "HOST"
+    }
+  },
+```
+
+For non-Docker applications, you don't need to specify anything.
+
+#### Specifying Ports
+
+You can specify the ports that are available through the `ports` array:
+
+```json
+    "ports": [
+        0, 0, 0
+    ],
+```
+
+In this example, we specify three randomly assigned ports which would then be available to our command via the environment variables `$PORT0`, `$PORT1` and `$PORT2`.
+
+These ports also become service ports for your application and are reserved cluster wide.
```

Comment:
Incorrect and therefore very confusing.

The ports that are assigned to your task as host ports are different from the service ports. (except with requirePorts where every last bit of sense of service ports falls apart).

With `requirePorts = false` (default) Marathon will only look at the port count to assign the correct number host ports to the task. It will assign unique service ports, though (at least unique for this Marathon instance) and return them via the API in this exact field. This is the horrible design decision that keeps on confusing scores of Marathon users.